### PR TITLE
Add delete msg api

### PR DESCRIPTION
**Pull request by simonhoss**

### CODE REVIEW

File: src/NATS.Client.JetStream/NatsJSContext.Streams.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public partial class NatsJSContext
     /// <exception cref="NatsJSException">There was an issue retrieving the response.</exception>
     /// <exception cref="NatsJSApiException">Server responded with an error.</exception>
     public ValueTask<NatsJSStream> CreateStreamAsync(string stream, string[] subjects, CancellationToken cancellationToken = default) =>
-        CreateStreamAsync(new StreamCreateRequest { Name = stream, Subjects = subjects }, cancellationToken);
+        CreateStreamAsync(new StreamCreateRequest {Name = stream, Subjects = subjects}, cancellationToken);
```
Comment (mtmk) [inline src/NATS.Client.JetStream/NatsJSContext.Streams.cs:18]: Please could you run `dotnet format` with your last commit and make sure there are no build warnings? Avoids having style changes polluting the diffs.

File: src/NATS.Client.JetStream/NatsJSContext.Streams.cs [hunk 2]
```diff
@@ -76,6 +76,27 @@ public partial class NatsJSContext
...
+    /// <summary>
+    /// Deletes a message from a stream.
+    /// </summary>
+    /// <param name="stream">Stream name to delete message from.</param>
+    /// <param name="request">Delete message request.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the API call.</param>
+    /// <returns>Delete message response</returns>
+    /// <exception cref="NatsJSException">There was an issue retrieving the response.</exception>
+    /// <exception cref="NatsJSApiException">Server responded with an error.</exception>
+    public async ValueTask<StreamMsgDeleteResponse> DeleteMessageAsync(
```
Comment (mtmk) [inline src/NATS.Client.JetStream/NatsJSContext.Streams.cs:88]: Should we implement a convenience method in the stream class as well?

**Review by mtmk — COMMENTED** [review]
Looks good. Thank you @simonhoss 🥇 

I have a couple of nitpicks if you have time please. Let me know if not, I can fix them in a later PR 😉

File: src/NATS.Client.JetStream/NatsJSContext.Streams.cs (hunk 1 above)
Comment (simonhoss) [reply src/NATS.Client.JetStream/NatsJSContext.Streams.cs]: done

File: src/NATS.Client.JetStream/NatsJSContext.Streams.cs (hunk 2 above)
Comment (simonhoss) [reply src/NATS.Client.JetStream/NatsJSContext.Streams.cs:88]: Good point. Done!

**Review by mtmk — APPROVED** [review]
LGTM

Thank you @simonhoss 👍

**Pull request merged**